Add v1.1 migration documentation (#325)
jellyfin/jellyfin-sdk-kotlin: docs/migration/v1.1.md

| jellyfin-core                 | jellyfin-core     |
| **jellyfin-platform-android** | **jellyfin-core** |
| jellyfin-api                  | jellyfin-api      |
| jellyfin-model                | jellyfin-model    |

Gradle will automatically use the Android build when the Android plugin is enabled. If you need to explicitly use the
Android build you can add the "-android" suffix.

## SDK initialization

The `Jellyfin {}` function to create an instance of the SDK was renamed to `createJellyfin {}`. The options set inside
of this builder changed depending on your platform. If the settings are incorrect it will throw an exception on
initialization.

### JVM

On the JVM there aren't many changes. Just make sure to rename `Jellyfin` to `createJellyfin`.

```kotlin
createJellyfin {
	clientInfo = ClientInfo(name = "My awesome client!", version = "1.33.7")
}
```

### Android

No longer needs the `android()` function called as this is automatically applied. Instead, set the context variable to
an Android context.

```kotlin
createJellyfin {
	context = /* Android Context */
	clientInfo = ClientInfo(name = "My awesome client!", version = "1.33.7")
}
```

## API extensions

> API interaction was improved for better performance and to make it easier to use. Previously when using the API you've

Reviewer: ```suggestion
API interaction was improved for better performance and to make it easier to use. Previously when using the API you
```

# Migrate to v1.1

When updating the SDK from version 1.0.z to 1.1.z you will need to make some changes to your code.
This document explains the changes required to build your application again with the new SDK
version.

## Package name changes

The SDK switched to Kotlin Multiplatform starting from v1.1. As a side effect some of our modules
have been moved around or renamed.

Each module now has separate builds for different platforms (common, jvm and android). Additionally

> the separate jellyfin-platform-android module was merged into jellyfin-core.0

Reviewer: ```suggestion
the separate jellyfin-platform-android module was merged into jellyfin-core.
```

The table below shows the new names to be used:

| Module (before)               | Module (after)    |
| ----------------------------- | ----------------- |
| jellyfin-core                 | jellyfin-core     |
| **jellyfin-platform-android** | **jellyfin-core** |
| jellyfin-api                  | jellyfin-api      |
| jellyfin-model                | jellyfin-model    |

Gradle will automatically use the Android build when the Android plugin is enabled.
If you need to explicitly use the Android build you can add the "-android" suffix.

## SDK initialization

> Creating an instance of the SDK got some changes. The `Jellyfin {}` function is now renamed to
> `createJellyfin {}`. The options set inside of this builder changed depending on your platform.

Reviewer: ```suggestion
The `Jellyfin {}` function to create an instance of the SDK was renamed to `createJellyfin {}`.
The options set inside of this builder changed depending on your platform.
```

> If the settings are incorrect they will throw an exception on start. So try running your app
> before publishing your changes!

Reviewer: ```suggestion
If the settings are incorrect it will throw an exception on initialization.
So try running your app before publishing your changes!
```

I guess you could also leave away the second sentence..

### JVM

> On the JVM there aren't much changes. Just make sure to rename `Jellyfin` to `createJellyfin`.

Reviewer: ```suggestion
On the JVM there aren't many changes. Just make sure to rename `Jellyfin` to `createJellyfin`.
```

```kotlin
createJellyfin {
	clientInfo = ClientInfo(name = "My awesome client!", version = "1.33.7")
}
```

### Android

No longer needs the `android()` function called as this is automatically applied.
Instead, set the context variable to an Android context.

```kotlin
createJellyfin {
	context = /* Android Context */
	clientInfo = ClientInfo(name = "My awesome client!", version = "1.33.7")
}
```

## API extensions

> Interacting with the API got some changes for better performance and to make it easier to use.

Reviewer: ```suggestion
API interaction was improved for better performance and to make it easier to use.
```

Previously when using the API you've had to follow two steps:

1. Create API client instance using `jellyfin.createApi`
2. Create instance of an API class that contains operations, like UserApi.

This came with a few issues. First, you had to keep track of those API classes to avoid constantly creating new ones.

> Secondly, you can't use autocomplete in an IDE to find the API classes and had to rely on api.jellyfin.org.

Reviewer: ```suggestion
Secondly, you couldn't use autocomplete in an IDE to easily find the available API classes and had to rely on api.jellyfin.org.
```
